### DISCUSSION TITLE
Detecting unused imports

### DISCUSSION THREAD

**Discussion by jamieQ** [peripheryapp/periphery#461] [Ideas]
hello! i've spent a bit of time recently looking into what it would take to add support for unused import detection. from what i understand, it seems like it should in theory be fairly straightforward – conceptually we should just need to add a pass that does something like the following:

1. walk all (non-import) declarations in a file
2. determine the source module for each declaration
3. compare this set to the set of imports (which is already computed via `ImportVisitor`)

however, i'm struggling to come up with a precise way to implement step `2.`. any thoughts on the general approach, or ideas about how to derive a mapping from symbols to the module providing their visibility?

**Comment by CraigSiemens** [comment]
This is something I'd like to see added as well. Our project is split into many frameworks so it'd be nice to know when one framework is no longer using code from another that it used to.
